### PR TITLE
Add LiquidatorProxy

### PR DESCRIPTION
**Pull request by BrendanChou**

### CODE REVIEW

File: contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+            setHeldWeiTo(cache.heldWei.add(heldWeiDelta), cache);
+            setOwedWeiTo(Types.zeroWei(), cache);
+        } else {
+            Types.Wei memory owedWeiDelta = heldWeiToOwedWei(cache.heldWei, cache);
+            setOwedWeiTo(cache.owedWei.add(owedWeiDelta), cache);
+            setHeldWeiTo(Types.zeroWei(), cache);
+        }
+    }
+
+    function figureOutWhereToStop(
```
Comment (AntonioJuliano) [inline contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol]: Can we make a more official name for this function haha?

Maybe `calculateLiquidationAmount` or something?

File: contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+            otherAccountId: 1,
+            data: new bytes(0)
+        });
+
+        // create accounts
+        Account.Info[] memory accounts = new Account.Info[](2);
+        accounts[0] = fromAccount;
+        accounts[1] = liquidAccount;
+
+        // execute the liquidations
```
Comment (AntonioJuliano) [inline contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol]: what's your thought on doing the liquidations in separate operations vs the same operations?

It seems like same operations would be less gas, and kind of conceptually makes sense.

Or is the problem you need the account balances to be updated by the time you consider the next liquidation?

Comment (BrendanChou) [reply contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol]: Yeah it's just way easier to do it this way TBH

File: contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol [hunk 3]
```diff
@@ -0,0 +1,446 @@
...
+
+
+/**
+ * @title LiquidatorProxyV1ForSoloMargin
+ * @author dYdX
+ *
+ * Contract for liquidating other accounts in Solo. Does not take marginPremium into account.
+ */
+contract LiquidatorProxyV1ForSoloMargin is
+    OnlySolo,
```
Comment (AntonioJuliano) [inline contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol]: Why is this needed? I don't see the `onlySolo` modifier used anywhere

Comment (AntonioJuliano) [reply contracts/external/proxies/LiquidatorProxyV1ForSoloMargin.sol]: Oh I see you're just using it for the `SOLO_MARGIN` state variable. I guess that's fine

File: package.json [hunk 4]
```diff
@@ -32,7 +32,7 @@
     "node": "ganache-cli",
     "debug": "truffle debug",
     "build:js": "rm -rf dist/ && tsc",
-    "docker_node": "ganache-cli -i 1313 -d -p 8545 -h 0.0.0.0 --db=/home/.ganache",
+    "docker_node": "ganache-cli -i 1313 -d -k -p 8545 -h 0.0.0.0 --db=/home/.ganache",
```
[the remark this answers is not in the document]
Comment (BrendanChou) [reply package.json]: hardfork version (Petersburg) the version that's actually on mainnet right now
